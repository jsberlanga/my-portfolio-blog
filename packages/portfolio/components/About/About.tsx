import * as React from 'react';
import { css } from '@emotion/react';
import RobotMe from '@juliosoto/components/Icons/RobotMe';
import SectionHeader from '../SectionHeader';
import { getMQ } from '@juliosoto/utils/styles';

const styles = {
  root: css`
    margin-bottom: var(--gap-bottom);

    .section-content {
      max-width: var(--content-width);
      line-height: 1.8em;
      margin-bottom: var(--gap-bottom);
    }
  `,
  info: css`
    display: flex;
    flex-direction: column;
    justify-content: space-around;
    > div {
      margin-bottom: var(--gap-bottom);
    }

    ${getMQ('desktop')} {
      flex-direction: row;
    }

    .info-skills {
      background-color: var(--c-background-02);
      color: var(--c-light-02);
      padding: var(--gap);
      width: 100vw;
      margin-left: calc(var(--gap) * -1);
      order: 0;

      ${getMQ('desktop')} {
        order: 1;
        width: 30%;
        margin-left: 2rem;
        padding: 2rem 3rem;
        border-radius: 4px;
      }
    }

    .info-experiences {
      order: 1;

      ${getMQ('desktop')} {
        order: 0;
        margin-right: 2rem;
        width: 70%;
      }
    }
  `,
};

const About: React.FC = () => {
  return (
    <div css={styles.root} id="about">
      <SectionHeader
        label={
          <React.Fragment>
            a bit about me <RobotMe />
          </React.Fragment>
        }
        subtitle="and how I ended up being a Software Engineer"
      />
      <p className="section-content">
        Lorem ipsum dolor sit amet consectetur adipisicing elit. Maiores
        quibusdam corporis id quisquam libero ad quo eos beatae nisi rem
        deserunt, placeat expedita distinctio, molestiae in debitis. Rerum
        possimus architecto provident, consequuntur dolore labore debitis{' '}
        ratione repudiandae, accusamus, sequi dolores itaque vitae aut beatae
        dolorum aliquam. Soluta excepturi necessitatibus corporis.
      </p>
      <div css={styles.info}>
        <div className="info-skills">
          <h3>skills</h3>
          <ul className="skills">
            <li>HTML / CSS / JavaScript</li>
            <li>TypeScript</li>
            <li>React</li>
            <li>Node</li>
            <li>Mongodb</li>
            <li>Amazon Web Services (AWS)</li>
            <li>Heroku</li>
            <li>Varnish Cache</li>
            <li>Datadog</li>
          </ul>
        </div>
        <div className="info-experiences">
          <h3>most relevant experience</h3>
          <ul className="experiences">
            <li>
              <p>
                <strong>Softwate Engineer</strong> at Schibsted Tech Polska
              </p>
              <p>
                - Lorem ipsum dolor sit amet consectetur adipisicing elit.
                Maxime commodi accusantium, reprehenderit sint molestias rerum
                praesentium quos tenetur nesciunt exercitationem nam.
                <br />- Lorem ipsum dolor sit amet consectetur adipisicing elit.
                Maxime commodi accusantium, reprehenderit sint molestias rerum
                praesentium quos tenetur nesciunt exercitationem nam.
              </p>
            </li>
            <li>
              <p>
                <strong>Cloud Engineer</strong> at Capgemini
              </p>
              <p>
                - As part of the Office 365 FastTrack Team I was responsible for
                the preparation and migration of clientsâ€™ data to the Microsoft
                cloud. From the onboarding process to the actual data migration
                and postmigration reporting.
              </p>
            </li>
          </ul>
        </div>
      </div>
    </div>
  );
};

export default About;
