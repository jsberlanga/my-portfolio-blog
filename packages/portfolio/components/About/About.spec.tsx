import * as React from 'react';
import { render, screen } from '@testing-library/react';
import About from './About';

jest.mock('@juliosoto/components/Icons', () => ({
  RobotMe: () => 'RobotMe',
}));
jest.mock('@juliosoto/lib/styles', () => ({
  getMQ: () => '500',
}));

describe('About', () => {
  test('renders correctly', () => {
    const { container } = render(<About />);

    expect(container).toMatchSnapshot();
    expect(screen.getByText('a bit about me RobotMe')).toBeInTheDocument();
  });
});
