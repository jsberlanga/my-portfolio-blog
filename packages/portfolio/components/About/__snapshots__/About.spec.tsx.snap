// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About renders correctly 1`] = `
.emotion-0 {
  margin-bottom: var(--gap-unit);
}

.emotion-0 .section-label {
  font-weight: 800;
  position: relative;
}

.emotion-0 .section-subtitle {
  font-size: 0.8rem;
  margin-bottom: var(--gap-unit-s);
}

.emotion-0 .section-subtitle 500 {
  font-size: 1rem;
}

<div>
  <div
    css="You have tried to stringify object returned from \`css\` function. It isn't supposed to be used directly (e.g. as value of the \`className\` prop), but rather handed to emotion so it can handle it (e.g. as value of \`css\` prop)."
    id="about"
  >
    <div
      class="emotion-0"
    >
      <h2
        class="section-label"
      >
        a bit about me 
        RobotMe
      </h2>
      <p
        class="section-subtitle"
      >
        and how I ended up being a Software Engineer
      </p>
    </div>
    <p
      class="section-content"
    >
      Hi there! First of all, thank you for being here! My name is Julio and originally I am from Granada, Spain. I came to KrakÃ³w some years ago, where I have lived and worked happily since.
      <br />
      <br />
      A couple of words about my professional life: I started my way into IT as an IT Operations Analyst working for
       
      <a
        class="link"
        href="https://www.hcltech.com/geo-presence/poland"
        rel="noreferrer"
        style="position: relative;"
        target="_blank"
      >
        HCL Poland
      </a>
       
      and
       
      <a
        class="link"
        href="http://kariera.sbdinc.pl/"
        rel="noreferrer"
        style="position: relative;"
        target="_blank"
      >
        Stanley Black and Decker Polska
      </a>
      . For personal reasons I moved back to KrakÃ³w where I got a job as a Cloud Engineer for
       
      <a
        class="link"
        href="https://capgemini.com"
        rel="noreferrer"
        style="position: relative;"
        target="_blank"
      >
        Capgemini Polska
      </a>
       
      on the Microsoft Team.
      <br />
      <br />
      In late 2018 I grew a love for web development. It started all of a sudden and since then, I knew that there was no turning back. I felt addicted to coding. I love it, every part of it, because it is amazingly challenging and fun.
      <br />
      <br />
      In September 2019,
       
      <a
        class="link"
        href="https://www.schibsted.pl/"
        rel="noreferrer"
        style="position: relative;"
        target="_blank"
      >
        Schibsted Tech Polska
      </a>
       
      gave me the opportunity to work a full time job as a FullStack Javascript Developer
       
      <span
        aria-label="tada"
        role="img"
      >
        ðŸŽ‰
      </span>
       
      It really has been a dream come true. And now I am lucky enough to everyday being able to work around TypeScript, React and Nodejs among other things.
    </p>
    <div
      css="You have tried to stringify object returned from \`css\` function. It isn't supposed to be used directly (e.g. as value of the \`className\` prop), but rather handed to emotion so it can handle it (e.g. as value of \`css\` prop)."
    >
      <div
        class="info-skills"
      >
        <h3>
          skills
        </h3>
        <ul
          class="skills"
        >
          <li>
            HTML / CSS / JavaScript
          </li>
          <li>
            TypeScript
          </li>
          <li>
            React
          </li>
          <li>
            Node
          </li>
          <li>
            GraphQL
          </li>
          <li>
            Jest
          </li>
          <li>
            Mongodb
          </li>
          <li>
            Amazon Web Services (AWS)
          </li>
          <li>
            Heroku
          </li>
          <li>
            Varnish Cache
          </li>
          <li>
            Datadog
          </li>
          <li>
            Terraform
          </li>
        </ul>
      </div>
      <div
        class="info-experiences"
      >
        <h3>
          relevant experience
        </h3>
        <ul
          class="experiences"
        >
          <li>
            <p>
              <strong>
                Softwate Engineer
              </strong>
               at Schibsted Tech Polska
            </p>
            <p>
              - As part of the Premium Shared Team we are responsible for delivering a platform that powers up the frontpages and articles of some of the biggest Scandinavia newspapers.
              <br />
              - We are responsible for creating tools and platforms that are easy for brand developers to create performant code. In our project we use TypeScript, React, Webpack, Node, Jest and Varnish cache. I also have the opportunity to build services using Amazon Web Services
            </p>
          </li>
          <li>
            <p>
              <strong>
                Cloud Engineer
              </strong>
               at Capgemini
            </p>
            <p>
              - As part of the Office 365 FastTrack Team I was responsible for the preparation and migration of clientsâ€™ data to the Microsoft cloud. From the onboarding process to the actual data migration and postmigration reporting.
            </p>
          </li>
        </ul>
      </div>
    </div>
  </div>
</div>
`;
